stonesoup/dataassociator/probability.py

```diff
@@ -137,19 +137,20 @@ def associate(self, tracks, detections, time):
 
             # record hypothesis for any given Detection being associated with
             # this track
-            for detection in detections:
+            for hypothesis in hypotheses[track]:
+                if not hypothesis:
+                    continue
                 pro_detect_assoc = Probability.sum(
                     joint_hypothesis.probability
                     for joint_hypothesis in joint_hypotheses
                     if joint_hypothesis.
-                    hypotheses[track].measurement is detection)
+                        hypotheses[track].measurement is hypothesis.measurement)
 
                 single_measurement_hypotheses.append(
                     SingleProbabilityHypothesis(
-                        hypotheses[track][0].prediction,
-                        detection,
-                        measurement_prediction=hypotheses[track][0].
-                        measurement_prediction,
+                        hypothesis.prediction,
+                        hypothesis.measurement,
+                        measurement_prediction=hypothesis.measurement_prediction,
```

Comment:
This is raising a flake8 error also. Easiest fix would be to change arguments to positional rather keyword arguments as the variables have good descriptive names (noting you'll need to switch _probability_ and _measurement_prediction_ arguments around).

Reply:
This may no longer be an issue following #166.